DigitalLearningSolutions.Data/DataServices/UserDataService/UserCentreDetailsDataService.cs

```diff
@@ -57,7 +57,15 @@ ELSE 0
                 transaction
             );
 
-            if (detailsAlreadyExist)
+            if (string.IsNullOrWhiteSpace(email))
+            {
+                connection.Execute(
+                    @"DELETE FROM UserCentreDetails WHERE userID = @userId AND centreID = @centreId",
+                    new { userId, centreId },
+                    transaction
+                );
+            }

```

Comment:
Could we include this in the "detailsAlreadyExist" if statement? We don't need to delete anything if the details don't exist. We would need to change things around a bit to make sure we aren't making null records in the first place.

Also it looks like this data service method doesn't have any unit tests. It would be good to implement some here.

Reply:
I don't think we should create the UserCentreDetails record when the email is null though.

Reply:
In which case, it would be good to keep the record with a null email. We should probably make sure the EmailVerified field is set to null as well.

Reply:
I added some tests, but it feels like I'm testing the functionality of SQL server, and I'm not sure if I can test something else about that method